mlrun/runtimes/base.py

```diff
@@ -1301,6 +1316,7 @@ def is_local(url):
 class BaseRuntimeHandler(ABC):
     # setting here to allow tests to override
     kind = "base"
+    class_modes: typing.Dict[typing.Union[RuntimeClassMode, str], str] = {}
```

Comment:
Why use a `Union` here? We have a nice enum, let's stick to the enum. Why do we also need a `str` as key?